projects/dotcms-field-elements/src/components/dot-textfield/dot-textfield.tsx

```diff
@@ -8,42 +8,106 @@ import Fragment from 'stencil-fragment';
 })
 export class DotTextfieldComponent {
     @Prop() value: string;
+    @Prop() name: string;
     @Prop() regexcheck: string;
+    @Prop() regexcheckmessage: string;
     @Prop() readOnly: string;
     @Prop() label: string;
     @Prop() hint: string;
     @Prop() placeholder: string;
     @Prop() required: boolean;
-    @Event() onChange: EventEmitter;
+    @Prop() requiredmessage: string;
+    @Prop() disabled: boolean;
+    @Event() valueChanges: EventEmitter;
+    @Event() statusChanges: EventEmitter;
 
     @State() _value: string;
     @State() _error = false;
+    @State() _dotTouched = false;
+    _dotPristine = true;
+
     _label: string;
 
     // tslint:disable-next-line:cyclomatic-complexity
-    validate(value: string): boolean {
-        if (this.required && value.length === 0) {
-            return true;
+    isValid(): boolean {
+        if (this.required && this._value.length === 0) {
+            return false;
         }
 
         if (this.regexcheck) {
             const regex = new RegExp(this.regexcheck, 'ig');
-            return !regex.test(value);
+            return regex.test(this._value);
         }
 
-        return false;
+        return true;
+    }
+
+    setTouched(): void {
+        if (!this._dotTouched) {
+            this._dotTouched = true;
+            this.emitStatusChange();
+        }
+    }
+
+    @Method()
+    reset(): void {
+        this._dotPristine = true;
+        this._dotTouched = false;
+        this._value = '';
+        this._error = false;
+        this.emitStatusChange();
     }
 
     // Todo: find how to set proper TYPE in TS
     setValue(event): void {
         this._value = event.target.value.toString();
-        this._error = this.validate(this._value);
-        this.onChange.emit({ error: this._error, value: this._value });
+        this._error = !this.isValid();
+        if (this._dotPristine) {
+            this._dotPristine = false;
+            this._dotTouched = true;
+        }
+        this.valueChanges.emit({ error: this._error, value: this._value, name: this.name });
+        this.emitStatusChange();
     }
 
-    componentWillLoad() {
+    emitStatusChange(): void {
+        this.statusChanges.emit({
+            name: this.name,
+            status: {
+                dotTouched: this._dotTouched,
+                dotValid: this.isValid(),
+                dotPristine: this._dotPristine
+            }
+        });
+    }
+
+    // tslint:disable-next-line:cyclomatic-complexity
+    errorMessage(): string {
+        return this.isValid()
+            ? ''
+            : this._dotPristine
+              ? ''
+              : this._value.length ? this.regexcheckmessage : this.requiredmessage;
+    }
+
+    componentWillLoad(): void {
         this._label = `dotTextfield_${generateId()}`;
```

Comment:
remove that function and now use the `name` which is unique on every custom element